diff --git CMakeLists.txt CMakeLists.txt
index 5dad9e9..972183d 100644
--- CMakeLists.txt
+++ CMakeLists.txt
@@ -51,6 +51,13 @@ add_library(rapidjson STATIC IMPORTED ${RAPIDJSON})
 # add subdirs
 
 add_subdirectory(src)
+
+# Suppress memcpy warnings from modern GCC/Clang in rapidjson.
+if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+    target_compile_options(discord-rpc PRIVATE "-Wno-class-memaccess")
+elseif(CMAKE_CXX_COMPILER_ID MATCHES "Clang|AppleClang")
+    target_compile_options(discord-rpc PRIVATE "-Wno-nontrivial-memcall")
+endif()
+
 if (BUILD_EXAMPLES)
     add_subdirectory(examples/send-presence)
 endif(BUILD_EXAMPLES)
